org.eclipse.jdt.debug.tests/tests/org/eclipse/jdt/debug/tests/breakpoints/SuspendVMConditionalBreakpointsTests.java

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ *  Copyright (c) 2022 Simeon Andreev and others.
+ *
+ *  This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License 2.0
+ *  which accompanies this distribution, and is available at
+ *  https://www.eclipse.org/legal/epl-2.0/
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ *
+ *  Contributors:
+ *     Simeon Andreev - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.debug.tests.breakpoints;
+
+import org.eclipse.debug.core.DebugEvent;
+import org.eclipse.debug.core.DebugException;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.model.IDebugTarget;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.debug.core.IJavaBreakpoint;
+import org.eclipse.jdt.debug.core.IJavaDebugTarget;
+import org.eclipse.jdt.debug.core.IJavaLineBreakpoint;
+import org.eclipse.jdt.debug.core.IJavaThread;
+import org.eclipse.jdt.debug.testplugin.DebugElementKindEventWaiter;
+import org.eclipse.jdt.debug.testplugin.DebugEventWaiter;
+import org.eclipse.jdt.debug.tests.AbstractDebugTest;
+import org.eclipse.jdt.debug.tests.TestUtil;
+
+/**
+ * Test that a SUSPEND_VM conditional breakpoints function as expected. Also tests for bug 575131.
+ */
+public class SuspendVMConditionalBreakpointsTests extends AbstractDebugTest {
+
+	/**
+	 * Constructor
+	 *
+	 * @param name
+	 */
+	public SuspendVMConditionalBreakpointsTests(String name) {
+		super(name);
+	}
+
+	/**
+	 * Tests that the VM is suspended when the a conditional line breakpoint is hit.
```

Comment:
Not sure I understood what the test actually does & when it should fail according to the comment one thread & one breakpoint are only needed. But I see **two** threads & **two** breakpoints? 
I guess the comment should be updated and explain that we test **race conditions** while setting VM suspending breakpoints.

Reply:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=575131#c3

Reply:
You're asked for review, I did it, and in order to understand the change I need explanation what the test is supposed to proof and how that is related to the bug.